Make tabLayout visible in the appbar after coming back from a searched order screen

rossanafmenezes opened the pull request:
Fixes #4374

## Changes 

Updates the showTabs function to display the tabLayout in the appbar when coming back from a searched order screen

## Screenshots (gifs)

| Before | After |
| ---- | ---- |
| <img src="https://user-images.githubusercontent.com/30724184/132672502-ed1b30cd-acec-44c5-aa88-30844aa34fcc.gif" width="350"/> | <img src="https://user-images.githubusercontent.com/30724184/132672545-dfb10975-e5eb-4f21-bfe4-c1c4a6bda655.gif" width="350"/> |

| Before | After |
| ---- | ---- |
| <img src="https://user-images.githubusercontent.com/30724184/132674720-3b601909-3c19-48ee-aba9-0fa7ad6f31bd.gif" width="350"/> | <img src="https://user-images.githubusercontent.com/30724184/132674730-ee005112-4986-4df5-a703-aa672b3a8887.gif" width="350"/> 

## Testing Instructions

**1**
- Go to the Orders tab
- Tap the search icon (top right).
- Enter a search term that shows at least one result.
- Tap on any search result.
- Tap the back arrow.
- Tap the back arrow again.
- Check to see the sub-navigation for "Processing" and "All Orders" is still present.

**2** 

- Enable do not keep activities
- Go to the Orders tab
- Tap the search icon (top right)
- send the app to the background
- Open the app again
- tap the back arrow once
- Check to see the sub-navigation for "Processing" and "All Orders" is still present.

- [ ] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

<!-- Pull request guidelines: https://github.com/woocommerce/woocommerce-android/blob/develop/docs/pull-request-guidelines.md -->

hichamboushaba reviewed (commented):
Thanks @rossanafmenezes for working on this, the solution you added works, but I believe we can simplify the logic further.
With the logic you added, I find it hard to understand what's going on, the `showTabs` function will always show the `tabLayout`, independent of the value of `show`, which means to fix the logic, we just don't need to hide it. So what I would suggest is remove the function `showTabs` completely and the code that calls it, because it seems it's not useful now, and we just overlook it when we did some refactoring.
To add to the reasons, if you check the line: https://github.com/woocommerce/woocommerce-android/blob/1af87a27cea59d023e510697398462c3969b4ff2/WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/list/OrderListFragment.kt#L537 here we remove the tabLayout when the search view is expanded, so showTabs doesn't match the behavior.
Please let me know what you think about this 🙂 

Another thing to keep an eye on, is to try to always create the branch for your PRs from an updated `develop` branch, this would avoid you having to fix lint rules in files that you didn't touch in this PR (I mean this commit: 5e612b445b825e86175bb4e3278b3426205c50bc), and would avoid the conflict it caused.
To fix this, I suggest the following commands:
```
git fetch
git reset --hard origin/develop
```
these two commands will reset your branch to be at the same point as `develop`, then make the changes you want, and then force-push them to update the PR.

PS: force pushing is generally not good when we are collaborating with other people, but since we agreed now about it, then it's OK 🙂.

rossanafmenezes committed:
Remove showTabs function

As suggested by the reviewer, showTabs function and the code calling it.
Also removed an unused import suggested in detektAll

rossanafmenezes commented:
Thank you @hichamboushaba for reviewing and for all the inputs 🙏 

I have followed your suggestions regarding the showTabs function and re-submitted it for review again.

hichamboushaba reviewed (approved):
This looks good to me, thanks @rossanafmenezes
